core/iwasm/fast-jit/fe/jit_emit_function.c

```diff
@@ -4,11 +4,229 @@
+        jitted_func_idx = func_idx - wasm_module->import_function_count;
+        GEN_INSN(LDI64, jitted_code, func_ptrs,
+                 NEW_CONST(I32, (uint32)sizeof(void *) * jitted_func_idx));
+    }
+#else
+    module_inst = jit_cc_new_reg_I32(cc);
+    GEN_INSN(LDI32, module_inst, cc->exec_env_reg,
+             NEW_CONST(I32, offsetof(WASMExecEnv, module_inst)));
+    if (func_idx >= wasm_module->import_function_count) {
+        module = jit_cc_new_reg_I32(cc);
+        func_ptrs = jit_cc_new_reg_I32(cc);
+        jitted_code = jit_cc_new_reg_I32(cc);
+        /* module = module_inst->module */
+        GEN_INSN(LDI32, module, module_inst,
+                 NEW_CONST(I32, offsetof(WASMModuleInstance, module)));
+        /* func_ptrs = module->fast_jit_func_ptrs */
+        GEN_INSN(LDI32, func_ptrs, module,
+                 NEW_CONST(I32, offsetof(WASMModule, fast_jit_func_ptrs)));
+        /* jitted_code = func_ptrs[func_idx - import_function_count] */
+        jitted_func_idx = func_idx - wasm_module->import_function_count;
+        GEN_INSN(LDI32, jitted_code, func_ptrs,
+                 NEW_CONST(I32, (uint32)sizeof(void *) * jitted_func_idx));
+    }
+#endif
+
+    if (func_idx < wasm_module->import_function_count) {
+        func_import = &wasm_module->import_functions[func_idx].u.function;
+        func_type = func_import->func_type;
+    }
+    else {
+        func = wasm_module
+                   ->functions[func_idx - wasm_module->import_function_count];
+        func_type = func->func_type;
+    }
+
+    /* Prepare parameters for the function to call */
+    outs_off =
+        cc->total_frame_size + offsetof(WASMInterpFrame, lp)
+        + wasm_get_cell_num(func_type->types, func_type->param_count) * 4;
+
+    for (i = 0; i < func_type->param_count; i++) {
+        switch (func_type->types[func_type->param_count - 1 - i]) {
+            case VALUE_TYPE_I32:
+#if WASM_ENABLE_REF_TYPES != 0
+            case VALUE_TYPE_EXTERNREF:
+            case VALUE_TYPE_FUNCREF:
+#endif
+                POP_I32(value);
+                outs_off -= 4;
+                GEN_INSN(STI32, value, cc->fp_reg, NEW_CONST(I32, outs_off));
+                break;
+            case VALUE_TYPE_I64:
+                POP_I64(value);
+                outs_off -= 8;
+                GEN_INSN(STI64, value, cc->fp_reg, NEW_CONST(I32, outs_off));
+                break;
+            case VALUE_TYPE_F32:
+                POP_F32(value);
+                outs_off -= 4;
+                GEN_INSN(STF32, value, cc->fp_reg, NEW_CONST(I32, outs_off));
+                break;
+            case VALUE_TYPE_F64:
+                POP_F64(value);
+                outs_off -= 8;
+                GEN_INSN(STF64, value, cc->fp_reg, NEW_CONST(I32, outs_off));
+                break;
+            default:
+                bh_assert(0);
+                break;
+        }
+    }
+
+    /* Commit sp as the callee may use it to store the results */
```

Comment:
To be clarified: the callee and the caller are using the one operand stack in `JitFrame` or not?